### PR TITLE
add opam

### LINKED ISSUE #1: Please make this plugin available on opam
Despite the claim to ["eventually submit as an opam package"](https://github.com/coq/coq/pull/714#issuecomment-307359498) a year and a half ago, I cannot find it [on opam](https://github.com/coq/opam-coq-archive/search?q=reduction-effects&unscoped_q=reduction-effects).  I think it is worth submitting now (and keeping up to date, and maybe adding to the CI?), even if it does not yet have the additional features you mentioned in that comment.

### PR DESCRIPTION
**Pull request by liyishuai**
resolve #1

### CODE REVIEW

**Comment by liyishuai** [comment]
I managed to compile the example. Where should I find the output?

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+clean:
+	if [ -e Makefile.coq ]; then $(MAKE) -f Makefile.coq $@; fi
+	$(RM) Makefile.coq* .*.aux
+
+test:
+	$(MAKE) -C tests clean
+	$(MAKE) -C tests
+
+Makefile.coq: _CoqProject
+	coq_makefile -f $< -o $@
```
Comment (JasonGross) [inline Makefile]:
```suggestion
	$(COQBIN)coq_makefile -f $< -o $@
```
Please use the version of Coq the user requests, not just the one in PATH

**Comment by JasonGross** [comment]
You should see the printing on stdout, I think.  (If not, I'd call it a bug.)  You should definitely see it if you do `cat $file.v | coqtop `.  I haven't tested this personally though.

Once you get the output working, I think you should make the test check that the output of running coqc on the test file matches expected output, and check in a .out or .out.expected file with the expected output.

Why "reduction-effect" rather than "reduction-effects" like the repo name?

**Comment by liyishuai** [comment]
Not quite working:
```sh
$ cat tests/PrintEffect.v | coqtop

Welcome to Coq 8.8.2 (January 2019)
Coq < Coq < Coq < [Loading ML file redeffect_plugin.cmxs ... done]

Coq <      = 3
     : nat

Coq <      = print_id (print_id (print_id 0))
     : nat

Coq <      = 0
     : nat

Coq <      = 6
     : nat

Coq <      = tt
     : unit

Coq <
```
I suppose we'd expect more output than the final result?

File: _CoqProject [hunk 2]
```diff
@@ -1,4 +1,3 @@
--R . ReductionEffect
```
Comment (liyishuai) [inline _CoqProject]: I have no idea why this line broke the compilation. Any suggestions?
Update: solved in https://github.com/coq/coq/issues/9846#issuecomment-477354824

**Pull request merged**